### PR TITLE
Feature/multistep form 2

### PR DESCRIPTION
**Pull request by expertdeveloper5821**
Currently need to check forget password and Sign-In with Google functionality.

### CODE REVIEW

**Comment by expertdeveloper5821** [comment]
Kindly review and merge this pr.

File: src/Components/CommonComponent/Navbar/Navbar.tsx [hunk 1]
```diff
@@ -5,20 +5,19 @@ import { useRouter } from 'next/navigation';
...
-  // const [, setError] = useState<string>('');
-  const [userData, setUserData] = useState<string>('');
-  const [nameData, setNameData] = useState<string>('');
+  const [userData, setUserData] = useState<string | undefined>('');
+  const [nameData, setNameData] = useState<string | undefined>('');
   const [initialsName, setInitialsName] = useState<string>('');
   const [pofile, setPofile] = useState<string | undefined>('');
+  const { triggerHandleLogout } = useUserContext();
 
   // function handleClosePopover() {
```
Comment (Ashok-tgc) [inline src/Components/CommonComponent/Navbar/Navbar.tsx]: remove comments

File: src/Components/CommonComponent/Navbar/Navbar.tsx [hunk 2]
```diff
@@ -27,8 +26,10 @@ export function Navbar() {
 
   const handleLogout = async () => {
     try {
```
Comment (Ashok-tgc) [inline src/Components/CommonComponent/Navbar/Navbar.tsx:22]: do we need tryCatch in this case?

File: src/Components/Forms/AuthForms/LoginForm.tsx [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+        try {
+          const response = await loginService({ email, password });
+          setIsLoading(false);
+          const decodedToken: DecodedToken = decodeJWt(response?.data?.userData?.token)!;
+          const date = new Date();
+          const expirationTime = date.setHours(date.getHours() + 1);
+
+          if (rememberMe) {
+            localStorage.setItem('rememberMe', rememberMe.toString());
```
Comment (Ashok-tgc) [inline src/Components/Forms/AuthForms/LoginForm.tsx]: line 50 and 53 are same, remove one

File: src/Components/Forms/AuthForms/LoginForm.tsx [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+      const url = new URL(window?.location?.href);
+      const extractedToken = url.searchParams.get('token');
+      if (extractedToken) {
+        handleVerifyTokenInLogin(extractedToken);
+      }
+    }
+  }, []);
+
+  useEffect(() => {
+    if (rememberMe) {
```
Comment (Ashok-tgc) [inline src/Components/Forms/AuthForms/LoginForm.tsx]: do we need this useEffect?

File: src/Components/Forms/AuthForms/LoginForm.tsx [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+        } else {
+          router.push('/auth/personaldetails');
+        }
+      } else if (decodedToken && decodedToken?.role?.role === 'admin') {
+        router.push('/adminDashboard/room');
+      } else if (decodedToken && decodedToken?.role?.role === 'spectator') {
+        router.push('/spectatorDashboard');
+      }
+    }
+  };
```
Comment (Ashok-tgc) [inline src/Components/Forms/AuthForms/LoginForm.tsx]: refactor this function

File: src/Components/Forms/AuthForms/PersonalDetailsForm.tsx [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+            name="player"
+            autoComplete="off"
+            placeholder="Player Id/Username"
+            value={values.player}
+            onChange={handleChange}
+            onBlur={handleBlur}
+          />
+
+          {/* <Tooltip
+            className={styles.toolTip}
```
Comment (Ashok-tgc) [inline src/Components/Forms/AuthForms/PersonalDetailsForm.tsx]: remove commented code everywhere

File: src/Components/Forms/AuthForms/ResetPasswordForm.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      const url = new URL(window?.location?.href);
+      const extractedToken = url.searchParams.get('token')!;
+      setToken(extractedToken);
+    }
+  }, []);
+
+  const initialValues: ResetFormValues = {
```
Comment (Ashok-tgc) [inline src/Components/Forms/AuthForms/ResetPasswordForm.tsx]: keep initailValues in one place in all files, in other files this variable in top, here inside component.

File: src/Components/Forms/AuthForms/ResetPasswordForm.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+          id="newPassword"
+          className={styles.password_wrapper}
+          name="newPassword"
+          autoComplete="off"
+          placeholder="Enter password"
+          value={values.newPassword}
+          onChange={handleChange}
+          onBlur={handleBlur}
+          onPaste={handlePaste}
+        ></Input>
```
Comment (Ashok-tgc) [inline src/Components/Forms/AuthForms/ResetPasswordForm.tsx]: use self close tag

File: src/Components/Forms/AuthForms/SignUpForm.tsx [hunk 9]
```diff
@@ -1,88 +1,88 @@
...
   const router = useRouter();
+  const [error, setError] = useState<string>('');
 
-  const initialValues: SignupFormValuesType = {
-    fullName: '',
-    userName: '',
-    email: '',
-    password: '',
-    upiId: '',
+  const handleRedirect = (token: string) => {
```
Comment (Ashok-tgc) [inline src/Components/Forms/AuthForms/SignUpForm.tsx]: if this function used in multiple files, create a common function and use it.

File: src/Components/Forms/AuthForms/SignUpForm.tsx [hunk 10]
```diff
@@ -1,88 +1,88 @@
...
+        router.push('/spectatorDashboard');
+      }
+    }
+    // else {
+    //   router.push('/auth/401');
+    // }
   };
 
-  const { values, touched, errors, handleSubmit, handleChange, handleBlur, setFieldValue } =
+  useEffect(() => {
```
Comment (Ashok-tgc) [inline src/Components/Forms/AuthForms/SignUpForm.tsx:26]: is it possible to create hoc for auth pages?

File: src/Components/Forms/AuthForms/SignUpForm.tsx [hunk 11]
```diff
@@ -1,88 +1,88 @@
...
-    if (storedEmail) {
-      setFieldValue('email', storedEmail);
-    }
-    if (storedPassword) {
-      setFieldValue('password', storedPassword);
-    }
-  }, [setFieldValue]);
   return (
-    <form onSubmit={handleSubmit}>
+    <form>
```
Comment (Ashok-tgc) [inline src/Components/Forms/AuthForms/SignUpForm.tsx:65]: is onSubmit missing?

File: src/Components/Forms/AuthForms/TeamsDetailsForm.tsx [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+          {isSubmitting ? 'Loading...' : <span className={styles.nextArrow}>Finish</span>}
+        </Button>
+
+        <Button className={styles.finish} onClick={() => router.push('/userDashboard')}>
+          <span className={styles.nextArrow}>Skip</span>
+        </Button>
+      </form>
+
+      {/* <div className={styles.flexGap}>
+        <div className={styles.rounded}></div>
```
Comment (Ashok-tgc) [inline src/Components/Forms/AuthForms/TeamsDetailsForm.tsx]: comments

File: src/Components/HOC/IsAuthenticatedHoc.tsx [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+      ) {
+        return <>{props.children}</>;
+      } else if (decodedToken?.role?.role === 'user' && !pathname.includes('userDashboard')) {
+        router.push('/auth/login');
+      } else if (decodedToken?.role?.role === 'admin' && !pathname.includes('adminDashboard')) {
+        router.push('/auth/login');
+      } else if (decodedToken?.role?.role === 'spectator' && !pathname.includes('spectator')) {
+        router.push('/auth/login');
+      }
+    }
```
Comment (Ashok-tgc) [inline src/Components/HOC/IsAuthenticatedHoc.tsx:28]: refactor this if possible

File: src/app/landingPage/termsCondition/page.tsx [hunk 14]
```diff
@@ -3,96 +3,85 @@ import styles from '@/styles/privacy.module.scss';
...
-        </li>
-      </ul>
-      <h3 className={styles.term_con_lastheading}>Cancellation & Refund Policy</h3>
-      <h3 className={styles.term_con_lastheading}>Last updated on Aug 1st 2023</h3>
-      <p>No cancellations & Refunds are entertained</p>
-  
-    </div>
+      <div className={styles.container2}>
+        <h1 className={styles.term_con_heading}>Terms & Conditions</h1>
+        <h2 className={styles.term_con_subheading}>Last updated on Aug 1st 2023</h2>
```
Comment (Ashok-tgc) [inline src/app/landingPage/termsCondition/page.tsx:9]: is it possible to create a array of objects and map this data?

File: src/app/userDashboard/video/page.tsx [hunk 15]
```diff
@@ -0,0 +1,69 @@
+'use client';
+import React, { useEffect, useState } from 'react';
+import styles from '@/styles/video.module.scss';
+import VideoCard from './videoCard/videoCard';
+// import CenterText from './videoCard/centerText';
```
Comment (Ashok-tgc) [inline src/app/userDashboard/video/page.tsx]: comments

File: src/app/userDashboard/video/page.tsx [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+                <option>2</option>
+                <option>3</option>
+              </select>
+            </div>
+          </div>
+          <div className={styles.container}>
+            <div className={styles.videoCard}>
+              <VideoCard onDataUpdate={handleChildData} />
+            </div>
+            {/* <div className={styles.centerText}>
```
Comment (Ashok-tgc) [inline src/app/userDashboard/video/page.tsx]: comments

File: src/app/userDashboard/video/videoCard/videoCard.tsx [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+  time: string;
+  videoLink: string;
+}
+interface CustomPaginationProps {
+  onDataUpdate: (data) => void;
+}
+
+const VideoCard: React.FC<CustomPaginationProps> = ({ onDataUpdate }) => {
+  const [data, setData] = useState<VideoInfo[]>([]);
+  // will use spectator login token here
```
Comment (Ashok-tgc) [inline src/app/userDashboard/video/videoCard/videoCard.tsx]: comments

File: src/app/userDashboard/video/videoCard/videoCard.tsx [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+  // console.log(data);
+  //   const accessToken =
+  // 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI2NGQyM2NmZmYyZGU4ZDVhODM2OTVlOWYiLCJyb2xlIjpbeyJfaWQiOiI2NGM3ODE1M2QyYzhhODQzMWNjMzZiZjIiLCJyb2xlIjpbImFkbWluIl19XSwiaWF0IjoxNjkxNzM2MzcwLCJleHAiOjE2OTE5MDkxNzB9.GGAIOjgZs9q82XdLZNvR-TQ4JwALiIev8lfLBtajhE4'
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const accessToken = localStorage.getItem('jwttoken');
+        const response = await sendRequest('/role/allvideolink', {
+          method: 'GET',
```
Comment (Ashok-tgc) [inline src/app/userDashboard/video/videoCard/videoCard.tsx:25]: can we add accessToken directly in sendRequest?

File: src/utils/contextProvider.tsx [hunk 19]
```diff
@@ -1,32 +1,125 @@
...
+  //       } else if (
+  //         decodedToken?.role?.role === 'admin' &&
+  //         !pathname.includes('adminDashboard') &&
+  //         (pathname.split('/')[1] === 'userDashboard' ||
+  //           pathname.split('/')[1] === 'spectatorDashboard')
+  //       ) {
+  //         router.push('/adminDashboard/room');
+  //       } else if (
+  //         decodedToken?.role?.role === 'user' &&
+  //         decodedToken?.upiId &&
```
Comment (Ashok-tgc) [inline src/utils/contextProvider.tsx]: comments

File: src/Components/CommonComponent/Navbar/Navbar.tsx (hunk 1 above)
Comment (expertdeveloper5821) [reply src/Components/CommonComponent/Navbar/Navbar.tsx]: done

File: src/Components/CommonComponent/Navbar/Navbar.tsx (hunk 2 above)
Comment (expertdeveloper5821) [reply src/Components/CommonComponent/Navbar/Navbar.tsx:22]: No not required, not worked on this part but, I have fixed this.

File: src/Components/Forms/AuthForms/LoginForm.tsx (hunk 3 above)
Comment (expertdeveloper5821) [reply src/Components/Forms/AuthForms/LoginForm.tsx]: done

File: src/Components/Forms/AuthForms/LoginForm.tsx (hunk 4 above)
Comment (expertdeveloper5821) [reply src/Components/Forms/AuthForms/LoginForm.tsx]: Removed this useEffect and adjusted logic

File: src/app/userDashboard/video/page.tsx (hunk 15 above)
Comment (expertdeveloper5821) [reply src/app/userDashboard/video/page.tsx]: done

File: src/app/userDashboard/video/page.tsx (hunk 16 above)
Comment (expertdeveloper5821) [reply src/app/userDashboard/video/page.tsx]: done

File: src/app/userDashboard/video/videoCard/videoCard.tsx (hunk 18 above)
Comment (expertdeveloper5821) [reply src/app/userDashboard/video/videoCard/videoCard.tsx:25]: done

File: src/utils/contextProvider.tsx (hunk 19 above)
Comment (expertdeveloper5821) [reply src/utils/contextProvider.tsx]: done

File: src/Components/Forms/AuthForms/LoginForm.tsx (hunk 5 above)
Comment (expertdeveloper5821) [reply src/Components/Forms/AuthForms/LoginForm.tsx]: did refactoring one scenario missing also fixed that.

File: src/Components/Forms/AuthForms/ResetPasswordForm.tsx (hunk 7 above)
Comment (expertdeveloper5821) [reply src/Components/Forms/AuthForms/ResetPasswordForm.tsx]: done.

File: src/Components/Forms/AuthForms/ResetPasswordForm.tsx (hunk 8 above)
Comment (expertdeveloper5821) [reply src/Components/Forms/AuthForms/ResetPasswordForm.tsx]: done.

File: src/Components/Forms/AuthForms/SignUpForm.tsx (hunk 9 above)
Comment (expertdeveloper5821) [reply src/Components/Forms/AuthForms/SignUpForm.tsx]: done.

File: src/Components/Forms/AuthForms/SignUpForm.tsx (hunk 10 above)
Comment (expertdeveloper5821) [reply src/Components/Forms/AuthForms/SignUpForm.tsx:26]: Already created AuthHoc.

File: src/Components/Forms/AuthForms/TeamsDetailsForm.tsx (hunk 12 above)
Comment (expertdeveloper5821) [reply src/Components/Forms/AuthForms/TeamsDetailsForm.tsx]: done.

File: src/Components/HOC/IsAuthenticatedHoc.tsx (hunk 13 above)
Comment (expertdeveloper5821) [reply src/Components/HOC/IsAuthenticatedHoc.tsx:28]: done.

File: src/app/landingPage/termsCondition/page.tsx (hunk 14 above)
Comment (expertdeveloper5821) [reply src/app/landingPage/termsCondition/page.tsx:9]: done.

File: src/Components/Forms/AuthForms/SignUpForm.tsx (hunk 11 above)
Comment (expertdeveloper5821) [reply src/Components/Forms/AuthForms/SignUpForm.tsx:65]: attached with button, Its working.

**Pull request merged**